FirebaseInstallations/Source/Library/FIRInstallations.m

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FIRInstallations.h"
+#import <FirebaseCore/FIRAppInternal.h>
+#import <FirebaseCore/FIRComponent.h>
+#import <FirebaseCore/FIRComponentContainer.h>
+#import <FirebaseCore/FIRLibrary.h>
+#import <FirebaseCore/FIRLogger.h>
+#import <FirebaseCore/FirebaseCore.h>
+
+#import "FIRInstallationsVersion.h"
+
+@protocol FIRInstallationsInstanceProvider
+@end
+
+@interface FIRInstallations () <FIRLibrary>
+@property(nonatomic, readwrite, strong) NSString *appID;
+@property(nonatomic, readwrite, strong) NSString *appName;
+@end
+
+@implementation FIRInstallations
+
+#pragma mark - Firebase component
+
++ (void)load {
+  [FIRApp registerInternalLibrary:(Class<FIRLibrary>)self
+                         withName:@"fire-install"
+                      withVersion:[NSString stringWithUTF8String:FIRInstallationsVersionStr]];
+}
+
++ (nonnull NSArray<FIRComponent *> *)componentsToRegister {
+  FIRComponentCreationBlock creationBlock =
+      ^id _Nullable(FIRComponentContainer *container, BOOL *isCacheable) {
+    *isCacheable = YES;
+    FIRInstallations *installations = [[FIRInstallations alloc] initWithApp:container.app];
```

Comment:
@ryanwilson  I would rather keep both `initWithGoogleAppID:appName:` and `initWithApp:` initializers, so the 2nd one uses the 1st. WDYT?

Reply:
Is there a particular reason you want to keep the `initWithApp:` initializer? I'm not sure what benefit there is outside of not having to change the initializer here in the component creation if it ever changes.

Reply:
I just would prefer to keep the logic (in this case extracting info from `FIRApp`) outside of the block used by a very specific method. If we can initialize an instance with a `FIRApp` having a convenience initializer sounds logical to me. 
Not too strong opinion here, I am fine with `initWithGoogleAppID:appName:` only if there are downsides on having both.

Reply:
@ryanwilson I pushed the version with 2 initializers just because I had it before your response. Let me know if you have a strong opinion to remove `initWithApp:`